test/puppetlabs/trapperkeeper/authorization/ring_middleware_test.clj

```diff
@@ -15,46 +20,287 @@
+                       :message "No certs found in PEM read from x-client-cert"]
+                      (cert-from-request "NOCERTSHERE"))))
+      (testing "fails when more than one cert is in the payload"
+        (let [cert-writer (StringWriter.)
+              _ (ssl-utils/cert->pem! testutils/test-domain-cert cert-writer)
+              _ (ssl-utils/cert->pem! testutils/test-domain-cert cert-writer)
+              certs-encoded (ring-codec/url-encode cert-writer)]
+          (is (thrown+? [:type :puppetlabs.trapperkeeper.authorization.ring-middleware/bad-request
+                         :message (str "Only 1 PEM should be supplied for "
+                                       "x-client-cert but 2 found")]
+                        (cert-from-request certs-encoded))))))))
 
 (def base-handler
-  (fn [request]
-    {:status 200 :body "hello"}))
+  (fn [_]
+    (ring-response/response "hello")))
 
 (defn build-ring-handler
-  [rules]
+  [rules allow-header-cert-info]
   (-> base-handler
-      (ring-middleware/wrap-authorization-check rules)))
-
-(deftest wrap-authorization-check-test
-  (logutils/with-test-logging
-    (let [ring-handler (build-ring-handler test-rule)]
-      (testing "access allowed when cert CN is allowed"
-        (let [response (ring-handler (testutils/request "/path/to/foo" :get "127.0.0.1" testutils/test-domain-cert))]
-          (is (= 200 (:status response)))
-          (is (= "hello" (:body response)))))
-      (testing "access denied when cert CN is not in the rule"
-        (let [response (ring-handler (testutils/request "/path/to/foo" :get "127.0.0.1" testutils/test-other-cert))]
-          (is (= 403 (:status response)))
-          (is (= (str "Forbidden request: www.other.org(127.0.0.1) access to "
-                      "/path/to/foo (method :get) (authentic: true) denied by "
-                      "rule 'test rule'.")
-                 (:body response)))))
-      (testing "access denied when cert CN is explicitly denied in the rule"
-        (let [response (ring-handler (testutils/request "/path/to/foo" :get "127.0.0.1" testutils/test-denied-cert))]
-          (is (= 403 (:status response)))
-          (is (= (str "Forbidden request: bad.guy.com(127.0.0.1) access to "
-                      "/path/to/foo (method :get) (authentic: true) denied by "
-                      "rule 'test rule'.")
-                 (:body response))))))
-    (testing "Denied when deny all"
-      (let [app (build-ring-handler [(-> (testutils/new-rule :path "/")
-                                         (rules/deny "*"))])]
-        (doseq [path ["a" "/" "/hip/hop/" "/a/hippie/to/the/hippi-dee/beat"]]
-          (let [req (testutils/request path :get "127.0.0.1" testutils/test-domain-cert)
-                {status :status} (app req)]
-            (is (= status 403))))))))
+      (ring-middleware/wrap-authorization-check rules allow-header-cert-info)))
+
+(deftest wrap-authorization-check-for-allow-header-cert-info-false-tests
```

Comment:
Most of the changes in here are just spacing - renamed from `wrap-authorization-check-test` to `wrap-authorization-check-for-allow-header-cert-info-false-tests`.  `build-ring-handler` now also called with extra argument, the `allow-header-cert-info` value.